### ISSUE TITLE
Model tests

### ISSUE THREAD

**Issue by jaentrouble** [jaentrouble/Adipose_tf_train#1]
First test : VGG-like
Result : Too complicated?

**Comment by jaentrouble** [comment]
Second test : smaller
conv32, same
max_pool
conv1, same

**Comment by jaentrouble** [comment]
## full_conv2_0
Model : full_conv2
lr : lr_step
epochs : 10
steps : 100

result : Overfitting?

**Comment by jaentrouble** [comment]
## full_conv3_0
Model : full_conv3
lr : lr_step
epochs : 10
steps : 100

result : no change at all

**Comment by jaentrouble** [comment]
## full_conv3_1
Model : full_conv3
lr : lr_step2 (slight modifed)

result : all becomes zero. something looks wrong

**Comment by jaentrouble** [comment]
## full_conv3_2
Model : full_conv3
lr : low_lr

__BUG_FOUND__
lr_update was not called

**Comment by jaentrouble** [comment]
## full_conv3_3

Model : full_conv3
lr : lr_step

result : no difference

**Comment by jaentrouble** [comment]
## Found Bug

last layer had ReLU activation -> making every output to 0
Changed to linear activation

**Comment by jaentrouble** [comment]
## full_conv3_4

Model : full_conv3_fix
lr : lr_step
epochs : 10
steps : 100

result : did show changes, but again, returned to 0

**Comment by jaentrouble** [comment]
## full_conv4_0
Model : full_conv4 -> added BatchNorm layer
lr : lr_step
epochs : 10
steps : 100

result : same...

**Comment by jaentrouble** [comment]
## full_conv4_1
Model : full_conv4_2 -> 2 BatchNorm layer
lr : lr_step
epochs : 10
steps : 100
*Nan occured -> switching lr_step to lr_step2*
![full_conv_4_1_val](https://user-images.githubusercontent.com/45917844/89551624-1cc1e180-d846-11ea-944f-6808eb8f7f54.png)

result : high valid accuracy; looks like learning rate was too high until last time?

**Comment by jaentrouble** [comment]
## full_conv4_2
Model : full_conv4_2
lr : lr_mul_inv
epochs : 10
steps : 100
![full_conv_4_2_val](https://user-images.githubusercontent.com/45917844/89553721-da4dd400-d848-11ea-8b6f-b863c67c06f3.png)

result : Not a huge change, looks like it works fine

**Comment by jaentrouble** [comment]
## full_conv5_0
Model : full_conv5 = lesser filters
lr : lr_mul_inv
epochs : 10
steps : 100

result:
![full_conv_5_0_val](https://user-images.githubusercontent.com/45917844/89658184-2fe9b580-d909-11ea-989e-5aa39b7223ae.png)

Bit dirty, but plausible? - need to check how much computation it costs in slow-computers

**Comment by jaentrouble** [comment]
## Comparing conv4_2 and conv5_0

### Conv4_2 > 0.25
![full_conv_4_2_val_025](https://user-images.githubusercontent.com/45917844/89659311-aa670500-d90a-11ea-89ab-d4bb9f96985f.png)

### Conv5_0 > 0.45
![full_conv_5_0_val_045](https://user-images.githubusercontent.com/45917844/89659602-0d589c00-d90b-11ea-892b-e2866232ac10.png)

**Comment by jaentrouble** [comment]
## ToDo
Try about 2x more complex model than full_conv4_2, and see if it is much better.
If not, move on to pruning the model.

**Comment by jaentrouble** [comment]
## full_conv_5_1
Model : full_conv5_2 -> twice more filters than 4_2
lr : lr_mul_inv
epochs = 10
steps = 100
### > 0.1
![full_conv_5_1_val_010](https://user-images.githubusercontent.com/45917844/90112873-87fe3d00-dd8b-11ea-9f4b-7403bc4a7df9.png)

result : very clean, but bit too much white
one more trial with longer training

**Comment by jaentrouble** [comment]
## full_conv_5_2
Model : full_conv5_2
lr : lr_mul_inv
epochs = 20
steps = 100
*Trained from scratch, not continuing from last model*
### >0.3
![full_conv_5_2_val_030](https://user-images.githubusercontent.com/45917844/90113795-d233ee00-dd8c-11ea-8632-a833fed9c10a.png)

result : not so different
Looks like 4_2 model was the gold spot.

**Comment by jaentrouble** [comment]
## Retrain with full_conv4_2
Model : full_conv4_2
lr : lr_mul_inv
epochs : 10
steps : 100

## >0.3
![full_conv_4_2_val_030_RE](https://user-images.githubusercontent.com/45917844/90116278-37d5a980-dd90-11ea-827a-a50cc84f212b.png)

Fix this model

**Comment by jaentrouble** [comment]
## Finish
Model to convert from raw image to binary mask
